packages/aws_bedrock/data_stream/invocation/elasticsearch/ingest_pipeline/default.yml

```
@@ -403,11 +405,21 @@ processors:
           .sorted()
           .collect(Collectors.toList());
 
-        ctx.gen_ai.policy.match_detail = nameAndDetailsOfPolicies.stream()
-          .filter(policy -> policy.details.match != null)
-          .map(policy -> policy.details)
-          .distinct()  // no sort because HashMap isn't Comparable
-          .collect(Collectors.toList());
+        ctx.gen_ai.policy.match_detail = Stream.concat(
+          nameAndDetailsOfPolicies.stream()
+            .filter(policy -> policy.details.match != null || policy.name == 'contextual_grounding_policy')
+            .map(policy -> policy.details),
+          allBodies.stream()
+            .filter(body -> body.trace?.guardrail?.outputAssessments instanceof List)
+            .flatMap(body -> body.trace.guardrail.outputAssessments.stream())
+            .filter(output -> output instanceof HashMap)
+            .flatMap(output -> output.entrySet().stream())
+            .map(assessment -> assessment.getValue())
+            .filter(details -> details.contextualGroundingPolicy?.filters instanceof List)
+            .flatMap(details -> details.contextualGroundingPolicy.filters.stream())
+        )
+        .distinct()  // no sort because HashMap isn't Comparable
+        .collect(Collectors.toList());
```

My diff was on `main` version of this file, so this part would just be:

```suggestion
        ctx.gen_ai.policy.match_detail = nameAndDetailsOfPolicies.stream()
          .filter(policy -> policy.details.match != null || policy.name == 'contextual_grounding_policy')
          .map(policy -> policy.details)
          .distinct()  // no sort because HashMap isn't Comparable
          .collect(Collectors.toList());
```

Currently the 2nd part of the concat generates nothing because it uses the camel case name `contextualGroundingPolicy`.

---

Thanks for clarifying, I missed that. Much nicer.